### ISSUE TITLE
Auto-completion is extremely slow

### ISSUE THREAD

**Issue by MathiasSven** [initialcommit-com/git-sim#76]
Hi, I found out about this project a couple of days ago, I wanted to help package it for my distro but while doing so I noticed two issues. 

The first one is pretty minor, when you run `git-sim --show-completion SHELL` the output is padded by `Manim Community v0.16.0.post0` using it for packaging is a little awkward, needing to pipe it through `sed '1,2d'`.

The second issue is that auto-completion is really slow, making it quite annoying to use. I have included a video showcasing it

https://user-images.githubusercontent.com/24759037/227434743-6bee5ce3-06bb-454d-b1d6-37960e805832.mp4

**Comment by initialcommit-io** [comment]
Hi again,

This is merged into dev now.  I'm putting the finishing touches on the next git-sim release so hopefully will have that out in a few days, including this enhancement.

I'll keep this open until it's deployed and you can verify it works.  I'll leave another note here when ready for you to test.

**Comment by initialcommit-io** [comment]
Just released this!  Please use git-sim version 0.2.9 to take advantage of this.  You can upgrade git-sim with: `pip install git-sim --upgrade`.

I'll close this for now since it's implemented, but please keep me posted on integrating git-sim into NixOS!

**Comment by initialcommit-io** [comment]
@MathiasSven Were you able to package this for NixOS?

**Comment by MathiasSven** [comment]
When I was trying to package `git-sim` I didn't notice there was an open [PR](https://github.com/NixOS/nixpkgs/pull/212680), but it had been stale for a while. 

I made changes to include auto-completion and to start at `0.2.7` rather than `0.1.2` but I was waiting on the PR author to include them. The PR seems to have gone inactive again, but it also wasn't building... I will update [my PR](https://github.com/SuperSandro2000/nixpkgs/pull/22) on the authors branch again since it is at 0.2.9 now.

**Comment by paketb0te** [comment]
@MathiasSven super cool workaround, thanks!

**Comment by initialcommit-io** [comment]
@MathiasSven Thanks for the update!  Hopefully it gets merged.  If you have a chance please let us know if it gets accepted!
